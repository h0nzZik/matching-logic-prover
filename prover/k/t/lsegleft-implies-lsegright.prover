claim \implies( \and( listSegmentLeft (variable("H") { ArrayIntInt } , variable("X") { Int }, variable("Y") { Int }, variable("F") { Set }, .Patterns), .Patterns )
              , \and( listSegmentRight(variable("H") { ArrayIntInt } , variable("X") { Int }, variable("Y") { Int }, variable("F") { Set }, .Patterns), .Patterns )
              )

strategy search-bound(5)

/*
strategy kt ; ( ( right-unfold ; simplify ; direct-proof )
              | ( simplify ; direct-proof )
              | kt ; ( (right-unfold-Nth(0, 1) ; right-unfold-Nth(0, 0) ; simplify ; direct-proof)
                     | (simplify ; direct-proof )
                     | (right-unfold-Nth(0, 1) ; simplify ; direct-proof)
                     )
              )
*/
