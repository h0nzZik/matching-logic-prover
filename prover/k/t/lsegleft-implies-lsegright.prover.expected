<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( lsegleft ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Y" ) { Int } , variable ( "F" ) { Set } , .Patterns ) , .Patterns ) , \and ( lsegright ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Y" ) { Int } , variable ( "F" ) { Set } , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
